### PR TITLE
Add support for new MongoDB store

### PR DESCRIPTION
**Pull request by OskarStark**
### TODO
* [x] Merge of https://github.com/php-llm/llm-chain/pull/42
* [x] `0.3` release if https://github.com/php-llm/llm-chain
* [ ] Merge this PR
* [ ] Create release of the bundle

cc @alcaeus

### CODE REVIEW

File: src/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -52,12 +52,15 @@ public function getConfigTreeBuilder(): TreeBuilder
                     ->useAttributeAsKey('name')
                     ->arrayPrototype()
                         ->children()
-                            ->enumNode('engine')->values(['chroma-db', 'azure-search'])->isRequired()->end()
+                            ->enumNode('engine')->values(['chroma-db', 'azure-search', 'mongodb'])->isRequired()->end()
```
[the remark this answers is not in the document]
Comment (chr-hertel) [reply src/DependencyInjection/Configuration.php:55]: yup, that's not nice, true

[the remark this answers is not in the document]
Comment (OskarStark) [reply src/DependencyInjection/Configuration.php:55]: See the other PR

File: src/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -52,12 +52,15 @@ public function getConfigTreeBuilder(): TreeBuilder
...
                             ->scalarNode('collection_name')->end()
                             ->scalarNode('api_key')->end()
                             ->scalarNode('endpoint')->end()
                             ->scalarNode('index_name')->end()
                             ->scalarNode('api_version')->end()
+                            ->scalarNode('database_name')->end()
+                            ->scalarNode('ur')->end()
+                            ->scalarNode('vector_field_name')->end()
```
Comment (OskarStark) [inline src/DependencyInjection/Configuration.php:63]: In a follow up PR; we can add validate calls to ensure what is needed for which engine

File: README.md [hunk 3]
```diff
@@ -49,6 +49,14 @@ llm_chain:
             endpoint: '%env(AZURE_SEARCH_ENDPOINT)%'
             index_name: '%env(AZURE_SEARCH_INDEX)%' 
             api_version: '2024-07-01'
+        mongodb:
+            engine: 'mongodb'
+            uri: '%env(MONGODB_URI)%'
```
Comment (chr-hertel) [inline README.md:54]: can be removed

File: src/DependencyInjection/Configuration.php [hunk 4]
```diff
@@ -52,12 +52,15 @@ public function getConfigTreeBuilder(): TreeBuilder
...
                         ->children()
-                            ->enumNode('engine')->values(['chroma-db', 'azure-search'])->isRequired()->end()
+                            ->enumNode('engine')->values(['chroma-db', 'azure-search', 'mongodb'])->isRequired()->end()
                             ->scalarNode('collection_name')->end()
                             ->scalarNode('api_key')->end()
                             ->scalarNode('endpoint')->end()
                             ->scalarNode('index_name')->end()
                             ->scalarNode('api_version')->end()
+                            ->scalarNode('database_name')->end()
+                            ->scalarNode('ur')->end()
```
Comment (chr-hertel) [inline src/DependencyInjection/Configuration.php:62]: ur i can be removed?

File: src/DependencyInjection/Configuration.php (hunk 2 above)
Comment (chr-hertel) [inline src/DependencyInjection/Configuration.php:63]: what about `collection_name`, `index_name` and `bulk_write`?

**Review by chr-hertel — APPROVED** [review]
yes please

**Comment by chr-hertel** [comment]
oof, more work than expected, but got it tagged :D

**Comment by OskarStark** [comment]
Thanks man

**Pull request merged**